src/classes/UTIL_OrgTelemetry.cls

```diff
@@ -85,10 +111,145 @@ public without sharing class UTIL_OrgTelemetry {
+     configuration.
+     */
+    private void handleOrgEnvironmentTelemetry() {
+        Integer activeCourseConnectionRecordTypes = 0;
+        Integer activeEthnicityPicklistValues = 0;
+        Boolean usingOldCourseDescription = false;
+
+        // Determine the number of active Course Connection record types.
+        Map<String, Schema.RecordTypeInfo> courseConnectionRecordTypesByDeveloperName = 
+                Schema.SObjectType.Course_Enrollment__c.getRecordTypeInfosByDeveloperName();
+        for (String developerName : courseConnectionRecordTypesByDeveloperName.keySet()) {
+            Schema.RecordTypeInfo rti = courseConnectionRecordTypesByDeveloperName.get(developerName);
+            if (rti.getDeveloperName() != 'Master' && rti.isActive()) {
+                activeCourseConnectionRecordTypes++;
+            }
+        }
+
+        // Determine the number of active Ethnicity picklist values.
+        Schema.DescribeFieldResult fieldResult = Contact.Ethnicity__c.getDescribe();
+        List<Schema.PicklistEntry> picklistEntries = fieldResult.getPicklistValues();
+        for (Schema.PicklistEntry pe : picklistEntries) {
+            if (pe.isActive()) {
+                activeEthnicityPicklistValues++;
+            }
         }
+
+        // Determine if the customer is using the new Extended Description field on the Course object.
+        // For this, we look at the 500 most recently modified Course records.
+        List<Course__c> courses = [SELECT Description__c, Extended_Description__c FROM Course__c ORDER BY LastModifiedDate DESC LIMIT 500];
+        for (Course__c course : courses) {
+            if (course.Description__c != null && course.Extended_Description__c == null) {
+                usingOldCourseDescription = true;
+                break;
+            }
+        }
+
+        featureManager.setPackageIntegerValue(TelemetryParameterName.Org_CountActiveCourseConnectionRecordTypes.name(), activeCourseConnectionRecordTypes);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.HasEDAInactiveTDTM.name(), 
+                [SELECT COUNT() FROM Trigger_Handler__c WHERE Owned_By_Namespace__c = 'hed' AND Active__c = false] > 0);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.HasUserManagedTDTM.name(), 
+                [SELECT COUNT() FROM Trigger_Handler__c WHERE User_Managed__c = true AND Active__c = true] > 0);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.UsingOldContactEthnicity.name(), activeEthnicityPicklistValues > 2);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.UsingOldCourseDescription.name(), usingOldCourseDescription);
+    }
+
+    /**
+     * @description Handles the calculation of all EDA feature telemetry that involves counting
+     * custom object record count.
+     */
+    private void handleObjectUsageTelemetry() {
```

Comment:
My only concern with this is that we may be trying process too many records at once and run into CPU limits.

I'm not concerned about hitting row limits, as the count() query only counts as one row queried. But for objects like Accounts, Course Connections, etc -- do we need to be worried about a high amount of data?

The nice thing is, since we're building this into separate batches already, it's really easy to break this out into additional batches if we need to. And we can implement counting similar to NPSP did for counting Opportunities. I just don't know if the queries against some of these objects will cause timeouts or not...

Reply:
If orgs do run into issues, the default value of these parameters is -1 (not 0), so we should be able to run a report in WLMA to determine if there are issues here.

Reply:
@lifewithryan I think yes, with additional work required. NPSP added an additional batch class that implements Database.Stateful to count with limits: https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls